[Net10] Invalidate empty view size on RecyclerView size change

kubaflo opened the pull request:
<!-- Please let the below note in for people that find this PR -->
> [!NOTE]
> Are you waiting for the changes in this PR to be merged?
> It would be very helpful if you could [test the resulting artifacts](https://github.com/dotnet/maui/wiki/Testing-PR-Builds) from this PR and let us know in a comment if this change resolves your issue. Thank you!

<!--
!!!!!!! MAIN IS THE ONLY ACTIVE BRANCH. MAKE SURE THIS PR IS TARGETING MAIN. !!!!!!! 
-->

### Description of Change

Refactored RecyclerViewHeight and RecyclerViewWidth to trigger InvalidateEmptyViewSize when their values change. This ensures the empty view is updated when the RecyclerView's dimensions are modified.

### Issues Fixed

<!-- Please make sure that there is a bug logged for the issue being fixed. The bug should describe the problem and how to reproduce it. -->

Fixes https://github.com/dotnet/maui/issues/31963

<!--
Are you targeting main? All PRs should target the main branch unless otherwise noted.
-->

kubaflo committed:
Invalidate empty view size on RecyclerView size change

Refactored RecyclerViewHeight and RecyclerViewWidth to trigger InvalidateEmptyViewSize when their values change. This ensures the empty view is updated when the RecyclerView's dimensions are modified.

jsuarezruiz reviewed (changes requested):
Could include a related test?

github-actions[bot] committed:
Update with new pipelines (#32265)

Co-authored-by: Rui Marinho <me@ruimarinho.net>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/dotnet build 20251030.4 (#32309)

On relative base path root
Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebAssembly , Microsoft.AspNetCore.Components.WebAssembly.Server , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.Extensions.Configuration , Microsoft.Extensions.Configuration.Abstractions , Microsoft.Extensions.Configuration.Json , Microsoft.Extensions.DependencyInjection , Microsoft.Extensions.DependencyInjection.Abstractions , Microsoft.Extensions.FileProviders.Abstractions , Microsoft.Extensions.Hosting.Abstractions , Microsoft.Extensions.Logging , Microsoft.Extensions.Logging.Abstractions , Microsoft.Extensions.Logging.Console , Microsoft.Extensions.Logging.Debug , Microsoft.Extensions.Primitives , Microsoft.JSInterop , Microsoft.NETCore.App.Ref From Version 10.0.0 -> To Version 10.0.0
Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.XUnitExtensions From Version 10.0.0-beta.25523.113 -> To Version 10.0.0-beta.25530.104
Microsoft.NET.Sdk From Version 10.0.100-rtm.25523.113 -> To Version 10.0.100-rtm.25530.104

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
Update dependencies from https://github.com/dotnet/dotnet build 20251031.2 (#32328)

On relative base path root
Microsoft.AspNetCore.Authentication.Facebook , Microsoft.AspNetCore.Authentication.Google , Microsoft.AspNetCore.Authentication.MicrosoftAccount , Microsoft.AspNetCore.Authorization , Microsoft.AspNetCore.Components , Microsoft.AspNetCore.Components.Analyzers , Microsoft.AspNetCore.Components.Forms , Microsoft.AspNetCore.Components.Web , Microsoft.AspNetCore.Components.WebAssembly , Microsoft.AspNetCore.Components.WebAssembly.Server , Microsoft.AspNetCore.Components.WebView , Microsoft.AspNetCore.Metadata , Microsoft.Extensions.Configuration , Microsoft.Extensions.Configuration.Abstractions , Microsoft.Extensions.Configuration.Json , Microsoft.Extensions.DependencyInjection , Microsoft.Extensions.DependencyInjection.Abstractions , Microsoft.Extensions.FileProviders.Abstractions , Microsoft.Extensions.Hosting.Abstractions , Microsoft.Extensions.Logging , Microsoft.Extensions.Logging.Abstractions , Microsoft.Extensions.Logging.Console , Microsoft.Extensions.Logging.Debug , Microsoft.Extensions.Primitives , Microsoft.JSInterop , Microsoft.NETCore.App.Ref From Version 10.0.0 -> To Version 10.0.0
Microsoft.DotNet.Arcade.Sdk , Microsoft.DotNet.Build.Tasks.Feed , Microsoft.DotNet.Build.Tasks.Installers , Microsoft.DotNet.Build.Tasks.Workloads , Microsoft.DotNet.Helix.Sdk , Microsoft.DotNet.RemoteExecutor , Microsoft.DotNet.SharedFramework.Sdk , Microsoft.DotNet.XUnitExtensions From Version 10.0.0-beta.25530.104 -> To Version 10.0.0-beta.25531.102
Microsoft.NET.Sdk From Version 10.0.100-rtm.25530.104 -> To Version 10.0.100-rtm.25531.102

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

dotnet-maestro[bot] committed:
[net10.0] Update dependencies from dotnet/macios (#32304)

* Update dependencies from https://github.com/dotnet/macios build 20251030.6
On relative base path root
Microsoft.iOS.Sdk.net10.0_26.0 , Microsoft.MacCatalyst.Sdk.net10.0_26.0 , Microsoft.macOS.Sdk.net10.0_26.0 , Microsoft.tvOS.Sdk.net10.0_26.0 From Version 26.0.11017 -> To Version 26.0.11018

* Update dependencies from https://github.com/dotnet/macios build 20251031.4
On relative base path root
Microsoft.iOS.Sdk.net10.0_26.0 , Microsoft.MacCatalyst.Sdk.net10.0_26.0 , Microsoft.macOS.Sdk.net10.0_26.0 , Microsoft.tvOS.Sdk.net10.0_26.0 From Version 26.0.11017 -> To Version 26.0.11019

* Update dependencies from https://github.com/dotnet/macios build 20251103.2
On relative base path root
Microsoft.iOS.Sdk.net10.0_26.0 , Microsoft.MacCatalyst.Sdk.net10.0_26.0 , Microsoft.macOS.Sdk.net10.0_26.0 , Microsoft.tvOS.Sdk.net10.0_26.0 From Version 26.0.11017 -> To Version 26.0.11020

---------

Co-authored-by: dotnet-maestro[bot] <dotnet-maestro[bot]@users.noreply.github.com>

rmarinho committed:
Update cgmanifest (#32360)

github-actions[bot] committed:
[api] Mark net10.0 GA API as shipped (#32363)

Co-authored-by: Rui Marinho <me@ruimarinho.net>

PureWeen commented:
/rebase
